build: add remote debugging with delve

arrayah opened the pull request:
## Description

This is a feat and adds functionality for a developer to remote-debug a testnet node. This is made possible with `delve` remote debugging.

I received some help from here: https://kupczynski.info/2020/05/17/remote-debug-go-code.html

Also, @creachadair I made a tiny modification in the testnet file generation. I removed the `localnet` folder and simply let the `testnet init-files` to use the default `.testnets`, since this is also being `.gitignore`d in the repo, and we don't usually want these files in the main repo.

Let me know if there are any implications to your update that you made a few days ago. 
---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [x] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [x] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [x] manually tested (if applicable)

arrayah commented:
@amaurym @marbar3778 is there any feedback on this PR? Anything I should do to move things further?

marbar3778 reviewed (approved):
generally looks good to me. Could we add some docs on how to use this?

arrayah commented:
> generally looks good to me. Could we add some docs on how to use this?

Where should I add it? 

There's just one single command:

```
make localnet-debug
```

Its the same thing like:

```
localnet-start
```

But the debug adds the delve-flavor docker image instead of the vanilla one.

amaurym commented on Makefile:
```diff
@@ -472,17 +477,24 @@ proto-update-deps:
-	$(if $(shell $(DOCKER) inspect -f '{{ .Id }}' cosmossdk/simd-env 2>/dev/null),$(info found image cosmossdk/simd-env),$(MAKE) -C contrib/images simd-env)
-	$(DOCKER) run --rm -v $(CURDIR)/localnet:/data cosmossdk/simd-env \
-		testnet init-files --v 4 -o /data --starting-ip-address 192.168.10.2 --keyring-backend=test
+localnet-build-env:
+	$(MAKE) -C contrib/images simd-env
+localnet-build-dlv:
```
Would love to have Makefile comments for what's the difference between those 2. Or these comments can go near `localnet-start` and `localnet-debug`

amaurym reviewed (commented):
I'm also fine with this. But I also would like some docs.

>> generally looks good to me. Could we add some docs on how to use this?

> Where should I add it?

Maybe just a `contrib/images/simd-dlv/README.md` for now? I'm also interested in what Delve is in general in that README.

arrayah commented:
Hello @amaurym and @marbar3778 I have updated the documentation with a readme file in the `contrib/images/simd-dlv/README.md` and updated comments in the `Makefile`.

amaurym reviewed (approved):
really nice, thanks!
